components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -390,6 +397,24 @@ public void handle(HttpServletRequest request, HttpServletResponse response, Aut
             }
             // Setting the authenticated user's object to the request to get the relevant details to log out the user.
             context.setProperty(FrameworkConstants.AUTHENTICATED_USER, authenticatedUser);
+            Flow flow = new Flow.Builder()
+                    .name(Flow.Name.LOGOUT)
+                    .initiatingPersona(Flow.InitiatingPersona.USER)
+                    .build();
+            IdentityContext.getThreadLocalIdentityContext().setFlow(flow);
+
+            try {
+                Optional<UserSession> userSession = FrameworkServiceDataHolder.getInstance()
+                        .getUserSessionManagementService()
+                        .getSessionBySessionId(authenticatedUser.getUserId(),
+                                context.getSessionIdentifier());
+                List<UserSession> userSessions = new ArrayList<>();
+                userSession.ifPresent(userSessions::add);
+                FrameworkUtils.publishUserSessionTerminateEvent(authenticatedUser, userSessions, false);
+            } catch (UserIdNotFoundException | SessionManagementException | IdentityEventException e) {
+                throw new FrameworkException("Error while getting user session.", e);
```

Comment:
Throwing this means, we are breaking the flow just because we cannot handle the event here. I think we just need to log the error and pass.